doc/ampi/manual.rst

```diff
@@ -213,79 +213,198 @@ Non-blocking collectives have been defined in AMPI since before
+underlying communication library one wants to use for running AMPI
+programs. See the charm/README file for details on picking the proper
+version. Here is an example of how to build a debug version of AMPI in a
+linux and ethernet environment:
 
-AMPI also provides extra communicator types that users can pass to
-``MPI_Comm_split_type``: ``AMPI_COMM_TYPE_HOST`` for splitting a
-communicator into disjoint sets of ranks that share the same physical
-host, ``AMPI_COMM_TYPE_PROCESS`` for splitting a communicator into
-disjoint sets of ranks that share the same operating system process, and
-``AMPI_COMM_TYPE_WTH``, for splitting a communicator into disjoint sets
-of ranks that share the same worker thread.
+.. code-block:: bash
 
-For parsing Fortran command line arguments, AMPI Fortran programs should
-use our extension APIs, which are similar to Fortran 2003’s standard
-APIs. For example:
+   $ ./build AMPI netlrts-linux-x86_64 -g
 
-.. code-block:: fortran
+And the following is an example of how to build a production version of
+AMPI on a Cray XC system, with MPI-level error checking in AMPI turned
+off:
 
-   integer :: i, argc, ierr
-   integer, parameter :: arg_len = 128
-   character(len=arg_len), dimension(:), allocatable :: raw_arguments
+.. code-block:: bash
 
-   call AMPI_Command_argument_count(argc)
-   allocate(raw_arguments(argc))
-   do i = 1, size(raw_arguments)
-       call AMPI_Get_command_argument(i, raw_arguments(i), arg_len, ierr)
-   end do
+   $ ./build AMPI gni-crayxc --with-production --disable-ampi-error-checking
 
-Name for Main Program
+AMPI can also be built with support for shared memory on any
+communication layer by adding "smp" as an option after the build target.
+For example, on an Infiniband Linux cluster:
+
+.. code-block:: bash
+
+   $ ./build AMPI verbs-linux-x86_64 smp --with-production
+
+AMPI ranks are implemented as user-level threads with a stack size
+default of 1MB. If the default is not correct for your program, you can
+specify a different default stack size (in bytes) at build time. The
+following build command illustrates this for an Intel Omni-Path system:
+
+.. code-block:: bash
+
+   $ ./build AMPI ofi-linux-x86_64 --with-production -DTCHARM_STACKSIZE_DEFAULT=16777216
+
+The same can be done for AMPI’s RDMA messaging threshold using
+``AMPI_RDMA_THRESHOLD_DEFAULT`` and, for messages sent within the same
+address space (ranks on the same worker thread or ranks on different
+worker threads in the same process in SMP builds), using
+``AMPI_SMP_RDMA_THRESHOLD_DEFAULT``. Contiguous messages with sizes
+larger than the threshold are sent via RDMA on communication layers that
+support this capability. You can also set the environment variables
+``AMPI_RDMA_THRESHOLD`` and ``AMPI_SMP_RDMA_THRESHOLD`` before running a
+job to override the default specified at build time.
+
+Building AMPI Programs
+----------------------
+
+AMPI provides compiler wrappers such as *ampicc*, *ampif90*, and
```

Comment:
maybe: `ampicc`   etc

Reply:
I don't think "`ampicc` etc" will be clear enough for first-time readers.

Reply:
I don't think "`ampicc` etc" will be clear enough for first-time readers.

Reply:
Sorry, what I meant here was, replacing `*ampicc* ` with ` ``ampicc`` `, `*ampif90* ` with ` ``ampif90`` `, and so on.

Reply:
Done